### PR TITLE
Update username whenever user logs in

### PR DESCRIPTION
**Pull request by amCap1712**
And with this change, we fully support username changes in LB . Earlier PRs migrated various components to use LB user id instead of username. With this, whenever a user logs in their username will be updated in LB so that website uses the new username as well.

### CODE REVIEW

File: listenbrainz/db/user.py [hunk 1]
```diff
@@ -515,21 +515,24 @@ def report_user(reporter_id: int, reported_id: int, reason: str = None):
         })
 
 
-def update_user_email(musicbrainz_id, email):
+def update_user_details(lb_id: int, musicbrainz_id: str, email: str):
     """ Update the email field for user with specified MusicBrainz ID
```
Comment (alastair) [inline listenbrainz/db/user.py:519]: update docstring

File: listenbrainz/db/user.py [hunk 2]
```diff
@@ -515,21 +515,24 @@ def report_user(reporter_id: int, reported_id: int, reason: str = None):
         })
 
 
-def update_user_email(musicbrainz_id, email):
+def update_user_details(lb_id: int, musicbrainz_id: str, email: str):
```
Comment (alastair) [inline listenbrainz/db/user.py:518]: does this method have a test? Looks like an easy win that we can add

Comment (amCap1712) [reply listenbrainz/db/user.py:518]: Added.

File: listenbrainz/db/user.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/db/user.py]: Done.

**Review by alastair — APPROVED** [review]

**Pull request merged**